cw-orch-daemon/src/json_file.rs

```diff
@@ -1,6 +1,86 @@
 use crate::DaemonError;
+use file_lock::{FileLock, FileOptions};
 use serde_json::{from_reader, json, Value};
-use std::fs::{File, OpenOptions};
+use std::{
+    fs::{File, OpenOptions},
+    io::Seek,
+};
+
+pub struct JsonFileState {
+    lock: FileLock,
+    json: Value,
+}
+
+impl JsonFileState {
+    /// Lock a new file
+    pub fn new(filename: &str) -> Self {
+        // open file pointer set read/write permissions to true
+        // create it if it does not exists
+        // dont truncate it
+
+        let options = FileOptions::new()
+            .create(true)
+            .read(true)
+            .write(true)
+            .truncate(false);
+
+        let lock: FileLock = FileLock::lock(filename, true, options).unwrap();
+
+        // return empty json object if file is empty
+        // return file content if not
+        let json: Value = if lock.file.metadata().unwrap().len().eq(&0) {
+            json!({})
+        } else {
+            from_reader(&lock.file).unwrap()
+        };
+
+        JsonFileState { lock, json }
+    }
+
+    /// Prepare json for further writes
+    pub fn prepare(&mut self, chain_id: &str, network_id: &str, deploy_id: &str) {
+        let json = &mut self.json;
+        // check and add network_id path if it's missing
+        if json.get(network_id).is_none() {
+            json[network_id] = json!({});
+        }
+
+        // add deployment_id to chain_id path
+        if json[network_id].get(chain_id).is_none() {
+            json[network_id][chain_id] = json!({
+                deploy_id: {},
+                "code_ids": {}
+            });
+        }
+    }
+
+    pub fn state(&self) -> Value {
+        self.json.clone()
+    }
+
+    /// Get a value for read
```

Comment:
unused

Reply:
> unused

Yes, as stated before, this is proof of concept, and not finished implementation.